packages/starknet-snap/src/utils/starknetUtils.ts

```diff
@@ -186,6 +186,21 @@ export const declareContract = async (
   });
 };
 
+export const getAccountNonce = async (
+  network: Network,
+  senderAddress: string,
+  privateKey: string | Uint8Array,
+  cairoVersion?: CairoVersion,
+): Promise<BigNumberish> => {
+  const nonceString = await getAccountInstance(
+    network,
+    senderAddress,
+    privateKey,
+    cairoVersion,
+  ).getNonce('latest');
+  return parseInt(nonceString.toLowerCase(), 16);
+};
+
 export const estimateFee = async (
```

Comment:
do we still need this method?

Reply:
We keep it for now because it's still used elsewhere.